### ISSUE TITLE
Dashboard broken after upgrade

### ISSUE THREAD

**Issue by Tonkonozhenko** [metabase/metabase#8590]
### Bugs
- Your databases: Redshift)
- Metabase version: master
- Metabase internal database: Postgres

I had a dashboard, then I upgraded metabase to master and I have the following error:

```
{:status :failed,
 :class clojure.lang.ExceptionInfo,
 :error
 "Must be a valid instance of one of these clauses: :and, :or, :not, :=, :!=, :<, :>, :<=, :>=, :between, :inside, :is-null, :not-null, :starts-with, :ends-with, :contains, :does-not-contain, :time-interval, :segment",
 :stacktrace
 ["query_processor.middleware.parameters.mbql$eval47016$build_filter_clause__47021.invoke(mbql.clj:38)"
  "query_processor.middleware.parameters.mbql$eval47016$build_filter_clause__47021$fn__47031$iter__47036__47040$fn__47041$fn__47042.invoke(mbql.clj:45)"
  "query_processor.middleware.parameters.mbql$eval47016$build_filter_clause__47021$fn__47031$iter__47036__47040$fn__47041.invoke(mbql.clj:44)"
  "query_processor.middleware.parameters.mbql$eval47016$build_filter_clause__47021$fn__47031.invoke(mbql.clj:44)"
  "query_processor.middleware.parameters.mbql$eval47016$build_filter_clause__47021.invoke(mbql.clj:38)"
  "query_processor.middleware.parameters.mbql$expand.invokeStatic(mbql.clj:71)"
  "query_processor.middleware.parameters.mbql$expand.invoke(mbql.clj:58)"
  "query_processor.middleware.parameters$expand_parameters_STAR_.invokeStatic(parameters.clj:20)"
  "query_processor.middleware.parameters$expand_parameters_STAR_.invoke(parameters.clj:14)"
  "query_processor.middleware.parameters$expand_parameters.invokeStatic(parameters.clj:45)"
  "query_processor.middleware.parameters$expand_parameters.invoke(parameters.clj:42)"
  "query_processor.middleware.parameters$substitute_parameters_STAR_.invokeStatic(parameters.clj:51)"
  "query_processor.middleware.parameters$substitute_parameters_STAR_.invoke(parameters.clj:48)"
  "query_processor.middleware.driver_specific$process_query_in_context$fn__46247.invoke(driver_specific.clj:12)"
  "query_processor.middleware.resolve_driver$resolve_driver$fn__49434.invoke(resolve_driver.clj:15)"
  "query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__44355$fn__44356.invoke(bind_effective_timezone.clj:9)"
  "util.date$call_with_effective_timezone.invokeStatic(date.clj:85)"
  "util.date$call_with_effective_timezone.invoke(date.clj:74)"
  "query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__44355.invoke(bind_effective_timezone.clj:8)"
  "query_processor.middleware.store$initialize_store$fn__51305$fn__51306.invoke(store.clj:11)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:32)"
  "query_processor.store$do_with_store.invoke(store.clj:28)"
  "query_processor.middleware.store$initialize_store$fn__51305.invoke(store.clj:10)"
  "query_processor.middleware.cache$run_query_and_save_results_if_successful_BANG_.invokeStatic(cache.clj:116)"
  "query_processor.middleware.cache$run_query_and_save_results_if_successful_BANG_.invoke(cache.clj:114)"
  "query_processor.middleware.cache$run_query_with_cache.invokeStatic(cache.clj:128)"
  "query_processor.middleware.cache$run_query_with_cache.invoke(cache.clj:124)"
  "query_processor.middleware.cache$maybe_return_cached_results$fn__44562.invoke(cache.clj:151)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__46134.invoke(catch_exceptions.clj:58)"
  "query_processor$process_query.invokeStatic(query_processor.clj:144)"
  "query_processor$process_query.invoke(query_processor.clj:140)"
  "query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:260)"
  "query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:254)"
  "query_processor$eval51350$process_query_and_save_execution_BANG___51355$fn__51356.invoke(query_processor.clj:284)"
  "query_processor$eval51350$process_query_and_save_execution_BANG___51355.invoke(query_processor.clj:270)"
  "api.card$run_query_for_card.invokeStatic(card.clj:583)"
  "api.card$run_query_for_card.doInvoke(card.clj:569)"
  "api.card$fn__58730$fn__58733.invoke(card.clj:590)"
  "api.card$fn__58730.invokeStatic(card.clj:589)"
  "api.card$fn__58730.invoke(card.clj:585)"
  "middleware$enforce_authentication$fn__54069.invoke(middleware.clj:117)"
  "api.routes$fn__65020.invokeStatic(routes.clj:61)"
  "api.routes$fn__65020.invoke(routes.clj:61)"
  "routes$fn__65109$fn__65110.doInvoke(routes.clj:108)"
  "routes$fn__65109.invokeStatic(routes.clj:103)"
  "routes$fn__65109.invoke(routes.clj:103)"
  "middleware$catch_api_exceptions$fn__54214.invoke(middleware.clj:427)"
  "middleware$log_api_call$fn__54190$fn__54192.invoke(middleware.clj:352)"
  "middleware$log_api_call$fn__54190.invoke(middleware.clj:351)"
  "middleware$add_security_headers$fn__54124.invoke(middleware.clj:254)"
  "core$wrap_streamed_json_response$fn__66172.invoke(core.clj:67)"
  "middleware$bind_current_user$fn__54074.invoke(middleware.clj:141)"
  "middleware$maybe_set_site_url$fn__54128.invoke(middleware.clj:278)"],
 :query
 {:type :query,
  :query
  {:source-table 78,
   :breakout [[:datetime-field [:field-id 3270] :day]],
   :aggregation [[:sum [:field-id 3266]]],
   :filter
   [:and
    [:!= [:field-id 3265] "a" "b" "c"]
    [:!= [:field-id 3267] "x" "y" "z"]
    [:!= [:field-id 3263] "1" "2" "3"]]},
  :constraints {:max-results 10000, :max-results-bare-rows 2000},
  :parameters [{:type "id", :target ["dimension" ["fk->" 3265 4575]], :value ["xxx"]} {:type "date/all-options", :target ["dimension" ["field-id" 3270]], :value "thismonth"}],
  :middleware nil,
  :cache-ttl 99,
  :info
  {:executed-by 1,
   :context :question,
   :card-id 10,
   :dashboard-id nil,
   :query-hash [47, 100, -80, 24, 15, 15, -70, 50, -113, 87, -82, -47, 112, -54, 78, -102, 99, 31, 34, 94, 93, 83, -9, -94, -32, -103, 126, -78, -85, 33, 17, -22],
   :query-type "MBQL"}},
 :expanded-query nil,
 :ex-data
 {:type :schema.core/error,
  :value [:= [:fk-> 3265 4575] "xxx"],
  :error
  (named [nil (named (named [nil (named (named (not (some-matching-condition? 3265)) "Must be a valid instance of one of these clauses: :field-id, :field-literal") "source-field") (named (named (not (some-matching-condition? 4575)) "Must be a valid instance of one of these clauses: :field-id, :field-literal") "dest-field")] "Must be a valid instance of one of these clauses: :field-id, :field-literal, :fk->, :datetime-field, :expression, :binning-strategy") "field") nil] "Must be a valid instance of one of these clauses: :and, :or, :not, :=, :!=, :<, :>, :<=, :>=, :between, :inside, :is-null, :not-null, :starts-with, :ends-with, :contains, :does-not-contain, :time-interval, :segment")}}
```

**Comment by camsaul** [comment]
Thanks @Tonkonozhenko I’ll take a look at this

**Comment by Tonkonozhenko** [comment]
Also when I tried to rollback complete dashboard was broken.
The issues was because in db card looked like `{"native": {"template-tags": ...}}` but not `{"native": {"template_tags": ...}}`

**Comment by camsaul** [comment]
Fixed by #8590
